### ISSUE TITLE
Column start (1) greater than line length (0)

### ISSUE THREAD

**Issue by bigslycat** [AtomLinter/linter-eslint#687]
![2016-09-08 13-00-38](https://cloud.githubusercontent.com/assets/1121997/18341661/d6546b3a-75cc-11e6-89f2-c29474fe713f.png)

```
Error: Column start (1) greater than line length (0)
  at Object.rangeFromLineNumber (/home/cat/.atom/packages/linter-eslint/node_modules/atom-linter/lib/index.js:179:11)
  at /home/cat/.atom/packages/linter-eslint/lib/main.js:142:33
  at Array.map (native)
  at /home/cat/.atom/packages/linter-eslint/lib/main.js:125:27
```
Atom v1.10.1, Ubuntu 16.10

**Comment by slackday** [comment]
I get the same error on Atom v.1.10.2, Mac OSX 10.10.5 after updating this plugin to v7.3.0 today.

Don't really know where to start looking for source of this error or what other info might be helpful.

**Comment by dev-pro-alexander-subota** [comment]
Same issue, Ubuntu 14.04, Atom 1.9.7, updated to plugin v7.3.0

**Comment by IanVS** [comment]
I am not seeing any problems with Atom `1.10.1`, mac OSX, and linter-eslint `7.3.0`.  

For those of you having this problem, can you try restarting Atom and confirm that it still occurs?  

If so, some more details would be great.  e.g. does this happen on every file you try to open, or only certain files?

**Comment by dev-pro-alexander-subota** [comment]
@IanVS It happens when user write the code and code have unclosed brackets.

**Comment by dev-pro-alexander-subota** [comment]
![scn-20160908-0160c75](https://cloud.githubusercontent.com/assets/16097870/18351119/8fa7e8e6-75e1-11e6-90ed-2293a49d319f.png)

**Comment by rockerest** [comment]
Also just updated to 1.10.2 and this issue persists.

For the record, I don't have to type anything, it happens on initial load of the editor, too.
Adding a blank line anywhere causes it. I don't think it has to do with unclosed braces.

![firstopen](https://cloud.githubusercontent.com/assets/277023/18360427/3a80b3a0-75bb-11e6-99d2-429a04bb7c50.png)

**Comment by slackday** [comment]
Weird and probably not too helpful in finding the cause of this issue but..

I was editing javascript files when this first occurred. File was correct and it didn't just on any specific input character just any typing. I had already tried restarting Atom before posting my first comment here and it didn't help. 

However after restarting my computer this error went away. Again on Atom v.1.10.2, Mac OSX 10.10.5. So I'm happy :)

**Comment by rockerest** [comment]
Just restarted to test, and the issue still persists after a reboot.

Running Xubuntu 16.04 (Ubuntu 16.04 base).

**Comment by halleycarleton** [comment]
I'm getting this error even after restarting atom. I had a similar issue with `linter-jshint` before I switched to `linter-eslint`. This latest update is the first time I have seen this issue with `linter-eslint`.

Here's a gif of the easiest way I have found to recreate the issue:
![recreating issue](http://i.imgur.com/BGlVPIX.gif)

Here is my [.eslintrc.json file](https://gist.github.com/halleycarleton/7de2e281aa86bfca4d11907d94d9dfaf/366f94858480e8db166e9ebda6b2dc8c6e662e0d).

```
linter-eslint 7.3.0
eslint        3.4.0
Ubuntu        16.04
```

**Comment by Arcanemagus** [comment]
Looking into this now.

**Comment by Arcanemagus** [comment]
@halleycarleton Your specific case is due to `no-multiple-empty-lines` reporting a column of 2 on blank lines... which don't have that many columns.

@dev-pro-alexander-subota and @rockerest it looks like you are both hitting the same thing as well.

@bigslycat I'm not sure what you are hitting as you provided no information necessesary to reproduce the error (file that shows it + config for `eslint` + version used), but I'm guessing you are hitting the same problem.

I'll get a nice clean reproduction case and file this over on ESLint.

**Comment by rockerest** [comment]
@Arcanemagus Can (probably) confirm that it's the `no-multiple-empty-lines` rule, causing other errors doesn't result in the popup, it looks like it's only errors when the line length is 0.

**Comment by Arcanemagus** [comment]
Filed over there, turns out this has been broken since `eslint@1` days, I have _no_ idea why this wasn't blowing up on everyone long ago 😕.

**Comment by lmj0011** [comment]
reverting back to `linter-eslint@7.2.4` is what I'm doing until a patch implemented

**Comment by Arcanemagus** [comment]
ESLint v3.5.0 includes the fix by @IanVS for this bug, update the local copy in your projects and you should stop seeing this error 😉.

**Comment by 22sk** [comment]
Nope, still facing the same issue.

```
> .\node_modules\.bin\eslint --version
v3.5.0
```

```
Error: Column start (1) greater than line length (0)
    at Object.rangeFromLineNumber (C:\Users\samuel\.atom\packages\linter-eslint\node_modules\atom-linter\lib\index.js:179:11)
    at C:\Users\samuel\.atom\packages\linter-eslint\lib\main.js:142:33
    at Array.map (native)
    at C:\Users\samuel\.atom\packages\linter-eslint\lib\main.js:125:27
```

**Comment by 22sk** [comment]
I don't even see eslint's path in the error trace. So why should eslint be responsible for the issue, or is the trace incomplete?

**Comment by scholtzm** [comment]
ESLint itself provides the invalid marker, e.g. column position 1 for line with 0 characters.

**Comment by 22sk** [comment]
But shouldn't the problem have been fixed with 3.5.0?

**Comment by IanVS** [comment]
@22sk, would you mind linting the file which is throwing the error using the eslint cli, and post the results?  It could well be another rule that is causing your problems.  The only one fixed in ESLint so far is `no-multiple-empty-lines`, but there are likely others as well.

**Comment by 22sk** [comment]
@IanVS The message appears like completely randomly, if any line length changes. The CLI's output looks just as it should. No thrown errors or such.

**Comment by Arcanemagus** [comment]
@22sk did you restart Atom after updating `eslint` in your project? It loads `eslint` per path, and won't update the loaded version until you restart Atom.

**Comment by jokeyrhyme** [comment]
I've restarted and I'm no longer getting the error related to `no-multiple-empty-lines`. Yay! Thanks!

I _did_ see it for one other error, but I can't remember how I reproduced it. When I figure it out, I'll report it against upstream ESLint.

**Comment by 22sk** [comment]
@Arcanemagus I did...

**Comment by scholtzm** [comment]
FWIW, ESLint v3.5.0 fixes the issue on my end.

**Comment by Arcanemagus** [comment]
v7.3.1 is released which should show the offending error in the message shown to you.

As the bug causing the originally reported issue has been fixed in `eslint@3.5.0` I'm locking this issue. Please file a _new_ issue here if you need help tracking down where the problem is, or preferably over on [ESLint](https://github.com/eslint/eslint/issues) as you should now be able to tell what the offending rule is 😉.

Thanks everyone for taking the time to help track this down and report it!
